src/AST/ClassExtensions.cs

```diff
@@ -54,99 +54,69 @@ public static Class GetNonIgnoredRootBase(this Class @class)
-                    select method).FirstOrDefault()
-                let rootBaseMethod = @base.Class.GetRootBaseMethod(@override, onlyFirstBase) ?? baseMethod
-                where rootBaseMethod != null || onlyFirstBase
-                select rootBaseMethod).FirstOrDefault();
+            return c.Bases
+                .Where(b => b.IsClass && b.Class.OriginalClass != c && (!onlyFirstBase || !b.Class.IsInterface))
+                .SelectMany(b => members(b.Class)
+                    .Where(@override.canOverride)
+                    .Select(m => b.Class.GetRootBaseMember(@override, members) ?? m)
+                    .Where(m => m != null || onlyFirstBase))
+                .FirstOrDefault();
         }
 
-        public static Property GetRootBaseProperty(this Class c, Property @override, bool onlyFirstBase = false)
+        public static Method GetRootBaseMethod(this Class c, Method @override, bool onlyFirstBase = false) 
         {
-            return (from @base in c.Bases
-                where @base.IsClass && @base.Class.OriginalClass != c && (!onlyFirstBase || !@base.Class.IsInterface)
-                let baseProperty = (
-                    from property in @base.Class.Properties
-                    where
-                        property.OriginalName == @override.OriginalName &&
-                        property.Parameters.SequenceEqual(@override.Parameters,
-                            new ParameterTypeComparer())
-                    select property).FirstOrDefault()
-                let rootBaseProperty = @base.Class.GetRootBaseProperty(@override, onlyFirstBase) ?? baseProperty
-                where rootBaseProperty != null || onlyFirstBase
-                select rootBaseProperty).FirstOrDefault();
+            return GetRootBaseMember(c, @override, (k => k.Methods), onlyFirstBase);
         }
 
-        public static Property GetPropertyByName(this Class c, string propertyName)
+        public static Property GetRootBaseProperty(this Class c, Property @override, bool onlyFirstBase = false) 
         {
-            Property property = c.Properties.FirstOrDefault(m => m.Name == propertyName);
-            if (property != null)
-                return property;
+            return GetRootBaseMember(c, @override, (k => k.Properties), onlyFirstBase);
+        }
+            
 
-            foreach (var baseClassSpecifier in c.Bases.Where(
-                b => b.Type.IsClass() && b.Class.IsDeclared))
-            {
-                property = baseClassSpecifier.Class.GetPropertyByName(propertyName);
-                if (property != null)
-                    return property;
-            }
-            return null;
+        public static M GetMemberByName<M>(this Class c, string memberName, Func<Class, IEnumerable<M>> members, Func<BaseClassSpecifier, bool> considerBase) where M : Declaration, IMember<M>
+        {
+            var member = members(c).FirstOrDefault(m => m.Name == memberName);
```

Comment:
You have changed the behaviour. Mine was depth-first, that is, the most basic member was returned. You need to flip the two parts of your new code: try the recursive call first and then the members.